src/pages/add.astro

```diff
@@ -4,98 +4,101 @@ import Button from '../components/Button.astro';
-    <span class="input-label">Speed (S)</span>
-    <number-input>
-      <NumberInput 
+    <div class="input-row">
+      <span class="input-label">Speed (s)</span>
+      <NumberInput
         id="speedInput"
-        placeholder="Enter numbers only"
+        placeholder="Enter numbers from 0.2-3"
         size="lg"
         minValue={0.02}
         maxValue={3}
         allowDecimals={true}
       />
-    </number-input>
-  </div>
-</div>
+    </div>
 
-<Button 
-  id="backButton"
-  size="lg"
-  backgroundColor="#4ea7f5"
-  textColor=white
-  borderColor="transparent"
-  href="/"
->
-  Back
-</Button>
+    <Button
+      id="backButton"
+      size="lg"
+      backgroundColor="#4ea7f5"
+      textColor="white"
+      borderColor="transparent"
+      href="/"
+    >
+      Back
+    </Button>
+
+    <!-- We'll handle client-side navigation for Next -->
+    <Button
+      id="nextButton"
+      size="lg"
+      backgroundColor="#4ea7f5"
+      textColor="white"
+      borderColor="transparent"
+    >
+      Next
+    </Button>
+  </div>
 
-<Button 
-  id="nextButton"
-  size="lg"
-  backgroundColor="#4ea7f5"
-  textColor=white
-  borderColor="transparent"
->
-  Next
-</Button>
+  <script type="module">
+    const nextBtn = document.querySelector('#nextButton');
+    const digitsInput = document.querySelector('#digitsInput');
+    const numbersInput = document.querySelector('#numbersInput');
+    const speedInput = document.querySelector('#speedInput');
 
-<script>
-  import { ProblemGenerator } from '../lib/ProblemGenerator';
-  const digitsInput = document.querySelector('#digitsInput') as HTMLInputElement;
-  const numbersInput = document.querySelector('#numbersInput') as HTMLInputElement;
-  const speedInput = document.querySelector('#speedInput') as HTMLInputElement;
-  const nextButton = document.querySelector('#nextButton');
+    nextBtn?.addEventListener('click', () => {
+      if (!digitsInput || !numbersInput || !speedInput) {
+        alert('Inputs not found in the DOM.');
+        return;
+      }
 
-  nextButton?.addEventListener('click', () => {
-    const digits = digitsInput?.value || '';
-    const numbers = numbersInput?.value || '';
-    const speed = speedInput?.value || '';
+      const digitsVal = digitsInput.value;
+      const numbersVal = numbersInput.value;
+      const speedVal = speedInput.value;
 
-    if (!digits || !numbers || !speed) {
-      alert('Please fill in all fields');
-      return;
-    }
+      if (!digitsVal || !numbersVal || !speedVal) {
+        alert('Please fill in all fields.');
+        return;
+      }
 
-    const gameConfig = {
-      digits: Number(digits),
-      numbers: Number(numbers),
-      speed: Number(speed)
-    };
+      // Build the URL to /flash with query params
+      const url = new URL('/flash', window.location.origin);
+      url.searchParams.set('digits', digitsVal);
+      url.searchParams.set('numbers', numbersVal);
+      url.searchParams.set('speed', speedVal);
```

Comment:
I'm not entirely sure this is correct: we will need to double-check when we move over to having Astro serve backend as well.